apps/emqx/src/emqx_channel.erl

```diff
@@ -1563,10 +1613,20 @@ handle_info({disconnect, ReasonCode, ReasonName, Props}, Channel) ->
 handle_info({puback, PacketId, PubRes, RC}, Channel) ->
     do_finish_publish(PacketId, PubRes, RC, Channel);
 handle_info({'DOWN', Ref, process, Pid, Reason}, Channel) ->
-    case emqx_hooks:run_fold('client.monitored_process_down', [Ref, Pid, Reason], []) of
-        [] -> {ok, Channel};
-        Msgs -> {ok, Msgs, Channel}
-    end;
+    ?EXT_TRACE_WITH_PROCESS_FUN(
+        broker_disconnect,
```

Comment:
Q: Why should we trigger the `broker_disconnect` event once we receive a `client.monitored_prcess_down` event?